### ISSUE TITLE
What support for `@override` tags?

### ISSUE THREAD

**Issue by Alexis-ROYER** [englercj/tsd-jsdoc#104]
I'm opening this issue in order to clarify what should be the appropriate behavior for `@override` tags.

As I'm working on making the tests pass on [PR#98](https://github.com/englercj/tsd-jsdoc/pull/98), I'm a bit confused with the output for the `MyThing.copy()` override in `test/fixtures/class_all.js` (see my commit [be91778](https://github.com/englercj/tsd-jsdoc/pull/98/commits/be917786fab390103a5b1085cad6f9a86297f4b6)).

According to [PR#11](https://github.com/englercj/tsd-jsdoc/pull/11), I understand that we should prioritize the parent's overridden doclet over the downclass's overriding one (which sounds weird to me).

But according to [PR#103](https://github.com/englercj/tsd-jsdoc/pull/103), it seems that the port to jsdoc@3.6.3 led us to prioritize the downclass's overriding doclet over the parent's overridden one (which seems more logical to me).
Indeed, jsdoc@3.6.3 raw output (`jsdoc -X`) differs from jsdoc@3.5.5, particularly with override doclets, especially with the final `module:util~MyThing#copy` doclets that receive the first `module:util~MyThing#copy` doclet in jsdoc@3.6.3 while it received the `OtherThing#copy` doclet in jsdoc@3.5.5.

If I get back to [jsdoc documentation](https://jsdoc.app/tags-override.html) for `@override` tags, I understand that `@override` tags could be considered as optional, in as much as jsdoc automatically detects overrides without the help of `@override`.
*Nevertheless*, when we remove the `@override` tag in `test/fixtures/class_all.js`, the raw jsdoc output is different:
- jsdoc@3.5.5:
    - The first `module:util~MyThing#copy` doclet:
        - no longer has a `"override": true` attribute,
        - no longer has a `"ignore": true` attribute,
        - but receives a `"overrides": "OtherThing#copy"` attribute.
    - The final `module:util~MyThing#copy` doclet is not generated anymore.
- jsdoc@3.6.6:
    - Same behavior for the first `module:util~MyThing#copy` doclet.
    - The final `module:util~MyThing#copy` docletS (two in a row) are not generated anymore.

On the typsecript side, it seems that in the [PR#103](https://github.com/englercj/tsd-jsdoc/pull/103) version of the `test/expected/class_all.d.ts` file, typescripts complains with the following error on `MyThing.copy()`:
```
Property 'copy' in type 'MyThing' is not assignable to the same property in base type 'OtherThing'.
  Type '(other: OtherThing) => void' is not assignable to type '() => void'.
```
So, even though the [PR#103](https://github.com/englercj/tsd-jsdoc/pull/103) way looked more logical, it seems that it eventually leads to invalid typescript.

In the end, I have the feeling that 'explicitly overridden members (i.e. a method that has a different signature)' as named in [PR#11](https://github.com/englercj/tsd-jsdoc/pull/11) should not be handled.
If so, the `@override` test case in `test/fixtures/class_all.js` could be removed.

**Comment by englercj** [comment]
`@override` is listed in the readme as an unsupported tag because there is no way we can declare a function of the same name, but different signature in a derived class in typescript currently.

As such (IIRC, because it was years ago) I was just skipping over anything that was inherited to try and generate valid TS. It wasn't that I was preferring the base class one, so much as I just couldn't define the child class version without generating invalid TS.

If the signatures are the same, you don't need to define it in the child because it is inherited. If the signatures are different, that is invalid TS. So I just dropped it.

I don't have strong feelings about how this *should* work. I think always generating the method, which might be invalid TS, is also OK. At least then the user knows they did something wrong.

**Comment by Alexis-ROYER** [comment]
Thank you very much for the clarification @englercj.

Then could we remove the `@override` tag from the `test/fixtures/class_all.js` test and align the signatures of the `copy()` methods in order to stay on the test case we actually support?

**Comment by englercj** [comment]
I'd like to have some test for the behavior of `@override` specifically to find where it breaks this lib.

While we can't support the actual meaning of `@override`, it would be nice if it behaved in a well-defined way.

**Comment by Alexis-ROYER** [comment]
Shall we say that, as long as TS does not support different signatures of overriding methods, the tsd-jsdoc should *not* generate types for overriding methods, whether these overriding methods have a different signature or not from the overridden ones?
In addition, we could display a warning when a `@override` tag is encountered.
Let me know.

**Comment by englercj** [comment]
Yes I think that is what the previous behavior was. If you inherit a method of the same name as one you have defined, we drop the child one.

I don't feel strongly about the override warning, I'm OK either way.

**Comment by Alexis-ROYER** [comment]
Thank you very much, that will help me for [PR#98](https://github.com/englercj/tsd-jsdoc/pull/98).

**Comment by Alexis-ROYER** [comment]
Fix proposal in [PR#98](https://github.com/englercj/tsd-jsdoc/pull/98) with commit [8b205c65](https://github.com/englercj/tsd-jsdoc/pull/98/commits/8b205c6579aa79d09bace3d8f26c10cb89cf28bb).
